Zend/tests/attributes/gh8421.phpt

```diff
@@ -0,0 +1,48 @@
+--TEST--
+Bug #8421: Attributes that target functions are not valid for anonymous functions defined within a method
+--FILE--
+<?php
+#[Attribute(Attribute::TARGET_FUNCTION)]
+class FunctionAttribute
+{
+    public int $number = 1;
+}
+
+$globalClosure       = #[FunctionAttribute] fn() => true;
+$globalStaticClosure = #[FunctionAttribute] static fn() => true;
+
+class ClosureHolder
+{
+    public function getClosureDefinedInScope(): Closure
+    {
+        return #[FunctionAttribute] fn() => true;
+    }
+
+    public function getStaticClosureDefinedInScope(): Closure
+    {
+        return #[FunctionAttribute] static fn() => true;
+    }
+
+    public static function getClosureDefinedInScopeStatically(): Closure
+    {
+        return #[FunctionAttribute] fn() => true;
+    }
+
+    public static function getStaticClosureDefinedInScopeStatically(): Closure
+    {
+        return #[FunctionAttribute] static fn() => true;
+    }
+}
+
+echo (new ReflectionFunction($globalClosure))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+echo (new ReflectionFunction($globalStaticClosure))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+echo (new ReflectionFunction(ClosureHolder::getClosureDefinedInScopeStatically()))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+echo (new ReflectionFunction(ClosureHolder::getStaticClosureDefinedInScopeStatically()))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+
+$holder = new ClosureHolder;
+
+echo (new ReflectionFunction($holder->getClosureDefinedInScope()))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+echo (new ReflectionFunction($holder->getStaticClosureDefinedInScope()))->getAttributes(FunctionAttribute::class)[0]->newInstance()->number;
+?>
+--EXPECT--
+111111
```

Comment:
I think I prefer `var_dump()` output here.

Anyhow, the trailing new line is missing.

Reply:
@cmb69 So expected output would be like the following?
```
int(1)
int(1)
int(1)
int(1)
int(1)
int(1)
```

Reply:
Yes; but I'm not really sure about that. Maybe the more compact form ("111111") is actually better. Perhaps just wait if others have opinions about this.

Reply:
Ah my bad, I jumped the gun and committed it. Will wait for other feedback, thanks.

Reply:
I think var_dump is perfectly fine. Diff output will give you a line number. Thus making it trivial to say the nth var_dump() fails instead of counting ones there.

Reply:
`var_dump` is fine with me too. But maybe you could even add an argument to the attribute (incrementing number) so if any of those cases didn't work we'd actually see which one it is.

Reply:
I can action that if you want @iluuu1994?

Reply:
Given the downvote Bob disagrees, I'm perfectly happy the way it is now.